build-system/tasks/check-types.js

```diff
@@ -258,6 +256,12 @@ async function typeCheck(targetName) {
 async function checkTypes() {
   const handlerProcess = createCtrlcHandler('check-types');
 
+  execOrThrow(
+    'npx -p typescript tsc --project ./src/compiler/tsconfig.json',
+    'Type checking compiler.js failed'
```

Comment:
Rather than special-casing this, can we add an option to the typecheck target definitions like `ts: true` or something and make this more generalized, rather than explicitly making it about the compiler here?